defender/wallet-migrator/autotask-1/index.js

```diff
@@ -0,0 +1,173 @@
+  const auth = {
+    username: covalentApiKey,
+  };
+
+  let responseData;
+
+  // get balance of all tokens sender address owns
+  try {
+    let response = await get(url, method, headers, auth);
+    responseData = response.data.data.items;
+  } catch (error) {
+    console.log('error retrieving wallet balance:');
+    console.log(error);
+  }
+
+  // create new provider and signer to perform on-chain transfers
+  console.debug('Creating DefenderRelayProvider');
+  const provider = new DefenderRelayProvider(autotaskEvent);
+  console.debug('Creating DefenderRelaySigner');
+  const signer = new DefenderRelaySigner(autotaskEvent, provider, { speed: 'fast' });
+  const relayerAddress = await signer.getAddress();
+
+  let erc20Contract;
+  let erc721Contract;
+
+  // these initial contract addresses are just placeholders - they will be replaced by each erc20 and erc721 contract address the user holds
+  // it can cause an error with ethers.js if the contract address is invalid, so we are passing in known valid contract addresses
+  // as place holders for erc20 and erc721 tokens on both goerli and mainnet
+  if (network === 'goerli') {
+    erc20Contract = new ethers.Contract('0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6', erc20Abi, signer);
+    erc721Contract = new ethers.Contract('0xCfD7cEF761A60dFBA0D240ee4fF82f7f51242675', erc721Abi, signer);
+  } else if (network === 'mainnet') {
+    erc20Contract = new ethers.Contract('0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2', erc20Abi, signer);
+    erc721Contract = new ethers.Contract('0xBC4CA0EdA7647A8aB7C2061c2E118A18a936f13D', erc721Abi, signer);
+  }
+
+  for (var i = 0; i < responseData.length; i++) {
+    const item = responseData[i];
+    const balance = item.balance;
+    const isNft = item.type === 'nft';
+    const isDust = item.type === 'dust'; // tokens with less than $0.1 in spot fiat value get classified as dust (ignore unless it's eth)
+    const decimals = item.contract_decimals;
+    const symbol = item.contract_ticker_symbol;
+    const isEther = item.contract_address === '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee';
+
+    // ignore if token value is less than $0.1 or is Ether
+    // address 0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee is used to represent Ether
+    // we cannot call allowance() or transferFrom() on Ether because it is not an erc20 token
+    // there isn't a way to approve an allowance of Ether ahead of time like we can or erc20 tokens
+    if (balance > 0 && !isNft && !isDust && !isEther) {
+      const specificErc20Contract = erc20Contract.attach(item.contract_address);
+      // check if relayer has allowance before sending
+      const allowance = await specificErc20Contract.allowance(senderWalletAddress, relayerAddress);
+      if (allowance > 0) {
+        const tx = await specificErc20Contract.transferFrom(senderWalletAddress, receiverWalletAddress, allowance);
```

Comment:
If allowance is higher than the balance, this will fail. Add a `Math.min(allowance, balance)` check to get the lowest of the 2

Reply:
converted balance to an ethers bignumber and compared it to allowance, then sent the min of the 2